pkg/backup/item_block_worker_pool.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright the Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	"context"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type ItemBlockWorkerPool struct {
+	inputChannel chan ItemBlockInput
+	wg           *sync.WaitGroup
+	logger       logrus.FieldLogger
+	cancelFunc   context.CancelFunc
+}
+
+type ItemBlockInput struct {
+	itemBlock  *BackupItemBlock
+	returnChan chan ItemBlockReturn
+}
+
+type ItemBlockReturn struct {
+	itemBlock *BackupItemBlock
+	resources []schema.GroupResource
+	err       error
+}
+
+func (p *ItemBlockWorkerPool) GetInputChannel() chan ItemBlockInput {
+	return p.inputChannel
+}
+
+func StartItemBlockWorkerPool(ctx context.Context, workers int, log logrus.FieldLogger) *ItemBlockWorkerPool {
+	// Buffer will hold up to 10 ItemBlocks waiting for processing
+	inputChannel := make(chan ItemBlockInput, 10)
```

Comment:
Let's add a variable for the input limit (10).

Reply:
I'm not sure we need a second configuration value here. This is just a buffer which sets how many inputs can be queued without blocking before a worker takes one. I can add it if we really want two configuration values here, but in most cases users won't really want to change this -- i.e. the same value that works with 1 worker (the default number of workers) is probably fine when they have 10 or 20 workers. So we can add it, but I don't expect tweaking it to change performance much (although setting it to a very small value (1 or 0) might negatively affect performance.